src/main/java/com/silviatanas/project/inventory/Main.java

```diff
@@ -1,23 +1,120 @@
+                            checkedFolder.mkdir();
+                            System.out.println("6");
+                        }
+
+                        // move processed files
+                        if (checkedFolder.isDirectory()) {
+                            System.out.println("7");
+                            Files.move(Paths.get(file.getAbsolutePath()), Paths.get(checkedFolder + File.separator + file.getName()), StandardCopyOption.REPLACE_EXISTING);
+                        }
+
+                        for (Environment e : contract.getEnvironments()) {
+                            System.out.println("8");
+
+                            // data written into csv
+                            writer.write(contract.getContractNumber() + ", ");
+                            writer.write(contract.getCustomerId() + ", ");
+                            writer.write(contract.getCustomerName() + ", ");
+                            writer.write(contract.getBillingSystemId() + ", ");
+                            writer.write(dateConvert.format(contract.getStartDate()) + ", ");
+                            writer.write(dateConvert.format(contract.getEndDate()) + ", ");
+
+                            writer.newLine();
+                            writer.write(e.name + ", ");
+                            writer.write(e.licenseInfo + ", ");
+                            writer.write(e.environmentType + ", ");
+                            writer.write(e.usageType + ", ");
+                            writer.write(e.swAGCloud + ", ");
+
+                            writer.newLine();
+                            e.setLicenseInfoObjects(parseLicenseInfo(e.licenseInfo));
+                            for (LicenseInfo l : e.licenseInfoObjects) {
+                                writer.write(l.getRow() + ", ");
+                                writer.write(l.getUnit() + ", ");
+                                writer.write(Arrays.toString(l.getProductCodes()));
+                                writer.newLine();
+                            }
+                            writer.write(Arrays.toString(e.licenseInfoObjects));
+
+                            writer.newLine();
+                            writer.write(Arrays.toString(contract.getRelatedContracts()));
+                            writer.newLine();
+                            writer.flush();
+                        }
+
+
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("Error writing into CSV");
+        }
+    }
+
+    public static LicenseInfo[] parseLicenseInfo(String info) {
+        Gson gson = new Gson();
+
+        LicenseInfo[] licenseInfo = gson.fromJson(info, LicenseInfo[].class);
+        return licenseInfo;
+    }
+
+    public static void main(String[] args) {
+        // setting folder paths
```

Comment:
Will be good if the src (Unchecked), target (Checked) folders and path to result (csv file) are configurable
There are 3 general ways configuration to be provided:
1. As main arguments - resived as values in args array passed to the main method
2. As system properties using System.getProperty(...);
3. Loading configuration from *.properties file using java.util.Properties class
Tipical flow should be
1. take configuration from main arguments if not presant 
2. take from System propery  if not present 
3. take from Propery, if not presant 
4. use the dafault vale (for example what is used in current iml)

Reply:
Configurations implemented